core/services/feeds/service.go

```diff
@@ -607,6 +636,16 @@ func (s *service) ProposeJob(ctx context.Context, args *ProposeJobArgs) (int64,
 	return id, nil
 }
 
+func isWFSpec(lggr logger.Logger, spec string) bool {
+	jobType, err := job.ValidateSpec(spec)
+	if err != nil {
+		// this should not happen in practice
```

Comment:
Seems like there's a lot of duplication around validating specs. I guess it's necessary for each function individually but it appears like it's getting called up to 4 times. Not a big deal but maybe we can refactor and consolidate a bit later